kafka-rest/src/main/java/io/confluent/kafkarest/controllers/TopicManager.java

```diff
@@ -53,8 +54,9 @@
   /**
    * Creates a new Kafka {@link Topic} with either partitions count and replication factor
    * or explicitly specified partition-to-replicas assignments.
+   * @return
    */
-  CompletableFuture<Void> createTopic(
+  CompletableFuture<CreateTopicsResult> createTopic(
```

Comment:
I agree with @dimitarndimitrov that we should not expose a `kafka.clients` type here in `kafkarest.controllers`. In particular, `CompletableFuture<CreateTopicsResult>` does not make sense, since `CreateTopicsResult` itself has another future inside. We flat-map all such cases in our interfaces. That all being said, it seems you do need to propagate some information about the topic creation up. I suggestion you creating a new auto-value in `kafkarest.entities` for this, e.g. `CreateTopicResult`, that only contain the information you need.